src/composable/orderTypes/Twap.ts

```diff
@@ -273,13 +273,21 @@ export class Twap extends ConditionalOrder<TwapData, TwapStruct> {
     }
 
     const cabinet = await this.cabinet(params)
-    const cabinetEpoch = utils.defaultAbiCoder.decode(['uint256'], cabinet)[0]
+    const rawCabinetEpoch = utils.defaultAbiCoder.decode(['uint256'], cabinet)[0] as BigNumber
+
+    // Guard against out-of-range cabinet epoch
+    if (rawCabinetEpoch.gt(MAX_UINT32)) {
+      throw new Error(`Cabinet epoch out of range: ${rawCabinetEpoch.toString()}`)
+    }
```

Comment:
Can we add a unit test for a cabinet overflow?

Reply:
Done.